### PR TITLE
Feature/backend auth service

### PR DESCRIPTION
**Pull request by linkouth**

### CODE REVIEW

**Review by usersadvocate — APPROVED** [review]

File: backend/src/modules/authentication/controllers/auth/auth.controller.ts [hunk 1]
```diff
@@ -2,27 +2,60 @@ import { Controller, Post, Body, Response, Request } from '@nestjs/common';
...
-
+  constructor(private readonly userService: UserService) {}
 
-    // TODO: call users service to create new user
+  @Post('signup')
+  async signUp(@Body() body: UserSignupDto, @Response() response: express.Response) {
+    const isValid = await this.userService.isValidSignupDto(body);
+    if (body && !isValid) {
+      response.status(401);
+      response.send({ status: 'error', message: 'Invalid form' });
```
Comment (MatvienkoDO) [inline backend/src/modules/authentication/controllers/auth/auth.controller.ts:19]: maybe after this there should be 'return;' for interrupting execution of function

File: backend/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "@nestjs/core": "^6.7.2",
     "@nestjs/mongoose": "^6.1.2",
     "@nestjs/platform-express": "^6.7.2",
-    "cookie-parser": "1.4.4",
+    "cookie-parser": "^1.4.4",
```
Comment (MatvienkoDO) [inline backend/package.json]: reason?

File: backend/src/modules/authentication/services/user/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    } else {
+      return false;
+    }
+  }
+
+  public async isUnique(signupDto: UserSignupDto) {
+    const user = await this.userModel.findOne({
+      email: signupDto.email
+    });
+    if (user) {
```
Comment (MatvienkoDO) [inline backend/src/modules/authentication/services/user/user.service.ts]: just ' return !user; '

File: backend/src/modules/authentication/services/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public async isValidSignupDto(signupDto: UserSignupDto): Promise<boolean> {
+    if (
+      signupDto.firstName &&
```
Comment (MatvienkoDO) [inline backend/src/modules/authentication/services/user/user.service.ts]: just ' return !!(... && ... && ...); '

File: backend/src/modules/authentication/controllers/auth/auth.controller.ts (hunk 1 above)
Comment (linkouth) [reply backend/src/modules/authentication/controllers/auth/auth.controller.ts:19]: response.send interrupt function execution

**Pull request merged**